packages/plugin-eslint/src/lib/runner/lint.ts

```diff
@@ -1,44 +1,101 @@
+  return withConfig(eslintrc, async configPath => {
+    // running as CLI because ESLint#lintFiles() runs out of memory
+    const { stdout } = await executeProcess({
+      command: 'npx',
+      args: [
+        'eslint',
+        `--config=${configPath}`,
+        '--no-eslintrc',
+        '--no-error-on-unmatched-pattern',
+        '--format=json',
+        ...toArray(patterns).map(pattern =>
+          // globs need to be escaped on Unix
+          platform() === 'win32' ? pattern : `'${pattern}'`,
+        ),
+      ],
+      ignoreExitCode: true,
+      cwd: process.cwd(),
+    });
+
+    return JSON.parse(stdout) as ESLint.LintResult[];
+  });
+}
+
+function loadRuleOptionsPerFile(
+  eslintrc: ESLintTarget['eslintrc'],
+  results: ESLint.LintResult[],
+): Promise<RuleOptionsPerFile> {
   const eslint = setupESLint(eslintrc);
 
-  const results = await eslint.lintFiles(patterns);
-
-  const ruleOptionsPerFile = await results.reduce(
-    async (acc, { filePath, messages }) => {
-      const filesMap = await acc;
-      const config = (await eslint.calculateConfigForFile(
-        filePath,
-      )) as Linter.Config;
-      const ruleIds = distinct(
-        messages
-          .map(({ ruleId }) => ruleId)
-          .filter((ruleId): ruleId is string => ruleId != null),
-      );
-      const rulesMap = Object.fromEntries(
-        ruleIds.map(ruleId => [
-          ruleId,
-          toArray(config.rules?.[ruleId] ?? []).slice(1),
-        ]),
-      );
-      return {
-        ...filesMap,
-        [filePath]: {
-          ...filesMap[filePath],
-          ...rulesMap,
-        },
-      };
-    },
-    Promise.resolve<RuleOptionsPerFile>({}),
-  );
+  return results.reduce(async (acc, { filePath, messages }) => {
+    const filesMap = await acc;
+    const config = (await eslint.calculateConfigForFile(
+      filePath,
+    )) as Linter.Config;
+    const ruleIds = distinct(
+      messages
+        .map(({ ruleId }) => ruleId)
+        .filter((ruleId): ruleId is string => ruleId != null),
+    );
+    const rulesMap = Object.fromEntries(
+      ruleIds.map(ruleId => [
+        ruleId,
+        toArray(config.rules?.[ruleId] ?? []).slice(1),
+      ]),
+    );
+    return {
+      ...filesMap,
+      [filePath]: {
+        ...filesMap[filePath],
+        ...rulesMap,
+      },
+    };
+  }, Promise.resolve<RuleOptionsPerFile>({}));
+}
 
-  return { results, ruleOptionsPerFile };
+async function withConfig<T>(
+  eslintrc: ESLintTarget['eslintrc'],
+  fn: (configPath: string) => Promise<T>,
+): Promise<T> {
+  if (typeof eslintrc === 'string') {
+    return fn(eslintrc);
+  }
+
+  const configPath = generateTempConfigPath();
+  await writeFile(configPath, JSON.stringify(eslintrc));
+
+  try {
+    return await fn(configPath);
+  } finally {
+    await rm(configPath);
+  }
+}
+
+function generateTempConfigPath(): string {
+  return join(
+    process.cwd(),
+    `.eslintrc.${Math.random().toString().slice(2)}.json`,
```

Comment:
❔ Do I understand it correctly that the limit for the number of projects is 999? That should probably be documented somewhere.

Reply:
Ah! I misunderstood the `slice`, thought it uses only first three digits not that it uses all digits after the first 3, hence the question. My bad. 😅